app.js

```diff
@@ -78,6 +80,33 @@ let Status = function(parent, index) {
 	this.tag.textContent = '🕑';
 };
 
+
+function checkViaImgSrc (imgUrl) {
+  // we check if gateway is up by loading 1x1 px image:
+  // this is more robust check than loading js, as it won't be blocked
+  // by privacy protections present in modern browsers or in extensions such as Privacy Badger
+  const imgCheckTimeout = 15000
+  return new Promise((resolve, reject) => {
```

Comment:
load events + timers + promises is some sticky business...

Reply:
indeed: entire `app.js` needs to be re-written from scratch in modern JS (painfully needs rate-limiting via execution queues, similar to ones we have in `js-libp2p-delegated-*`)

@jessicaschilling just like with cid.ipfs.io, we should rewrite this app (both backend and frontend) at some point (2021 Q1/Q2-ish?)

Reply:
@lidel There's a long tail of improvements to the gateway checker that never got completed here: https://github.com/ipfs/public-gateway-checker/issues/93

I'd suggest seeing what of that could be salvaged, or at least using that as the base issue for a rewrite. I'll add the notes from this PR to that issue.